src/basic/terminal-util.c

```diff
@@ -278,6 +278,8 @@ int reset_terminal_fd(int fd, bool switch_to_text) {
                 goto finish;
         }
 
+        (void) loop_write(fd, "\033[?7h", 5); /* Enable line wrapping */
```

Comment:
hmm, yuck. sucks we have to do this.

also kinda sucks that this is an ANSI sequence and not an ioctl.

But I have the suspicion that we really have to do this here.

But could you please move this after the tcsetattr() below, i.e. at a point where the tty driver itself is already in a good known state? I think writing to a device is something we should only try if we know the tty is somewhat OK set up in its basic wiring.

Reply:
hmm, we probably should not send ansi sequence if we are told to operate in dumb mode. hence condition on terminal_is_dumb()?

Reply:
hmm, and then i am a bit afraid of the line being dead and we block on it for good. Hence, can you please use loop_write_full() here and specify a timeout?

and please switch the device to O_NONBLOCK first, and reset afterwards.

maybe extend fd_nonblock() for that, by making it return 0 if it didn't change anything, and 1 if it did, so that we know whether we need to turn it off again afterwards or not.

Reply:
also, instead of specifying 5 as length, pass SIZE_MAX, so that the call does strlen() internally